### PR TITLE
Load examiner.js when #development=2

### LINKED ISSUE #9326: Introduce examiner.js
Following discussion [here](https://github.com/ampproject/amphtml/pull/9260).

`examiner.js` is a js tool that's served at `https://cdn.ampproject.org/examiner.js` and executed when user turn on `#development=1` on an AMP page. 

It prints extra debug info to developers, in addition to `validator.js`, which prints format rules violations.

A main use case can be to leverage browser API [PerformanceObserver](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver) to monitor what 3P frames is exhausting resources and slowing things down.

@dvoytenko

### PR DESCRIPTION
**Pull request by lannka**
closes #9326

### CODE REVIEW

File: src/validator-integration.js [hunk 1]
```diff
@@ -34,14 +35,24 @@ export function maybeValidate(win) {
   }
 
   /** @const {!Element} */
-  const s = win.document.createElement('script');
+  const validatorScript = win.document.createElement('script');
```
Comment (dvoytenko) [inline src/validator-integration.js]: Please remove the TODO on the next line.

File: src/validator-integration.js [hunk 2]
```diff
@@ -34,14 +35,24 @@ export function maybeValidate(win) {
...
-    /* global amp: false */
-    amp.validator.validateUrlAndLog(filename, win.document, getMode().filter);
-  };
-  win.document.head.appendChild(s);
+  validatorScript.src = `${urls.cdn}/v0/validator.js`;
+
+  const examinerScript = win.document.createElement('script');
+  examinerScript.src = `${urls.cdn}/examiner0.js`;
+
+  Promise.all([
```
Comment (dvoytenko) [inline src/validator-integration.js]: Worth delaying the validation before examiner is loaded? Maybe better split into separate calls? I.d. probably factor out a shared function, e.g.:
```
function loadScript(src) {
  const el = ...
  const promise = loadPromise(el);
  promise.catch().then(() => head.removeChild(el));
  head.appendChild(el);
  return promise;
}

loadScript(validator).then(() => {validateUrlAndLog()})
loadScript(examiner).then(() => ...)
```

?

Comment (lannka) [inline src/validator-integration.js]: the examiner.js script might fail the validator. we need to remove it from DOM before run validate()

Comment (dvoytenko) [inline src/validator-integration.js]: I see. Ok. We'll have to live with `Promise.all()`. Please also consider an alternative to wait to load examiner until validator is done, though that might be too long...

**Comment by lannka** [comment]
The original idea is from this PR: #9260, where I wanted to monitor the long tasks in those 3P iframes on our ads.amp.html example page, so I can have an idea how a new ad network performs.

@dvoytenko and I then were talking about how we can use the data. I figured this might be something publishers didn't pay attention but could be interested to know, although I'm not sure if there is immediate actions they can take on them. Console log is the easiest way to surface such data. Later on, if found useful, we can bring in amp-analyltics support.

Are you concerning about the overhead it brings? If so, we can make it development=2, but then it's less discoverable.

> What should they do? File an AMP bug. 

This only reports slowness from 3p iframes. See example:

3ec1034e-35d1-11e7-9e9c-578cae4a943f

**Comment by lannka** [comment]
Also worth to mention that in one of the meeting, MOAT was asking if there is any tooling in AMP to make sure their 3P iframe meets AMP's requirements (in any). This can be a start point.

**Comment by lannka** [comment]
@cramforce as discussed offline, made it `#development=2` PTAL

File: src/mode.js [hunk 3]
```diff
@@ -110,6 +110,7 @@ function getMode_(win) {
     // Triggers validation
     development: !!(hashQuery['development'] == '1' ||
         win.AMP_DEV_MODE),
+    examiner: hashQuery['development'] == '2',
```
Comment (cramforce) [inline src/mode.js]: Magic :)

**Pull request merged**